collector/database/classes/task/metrics_cleanup_task.php

```diff
@@ -36,7 +37,16 @@ public function get_name() {
     public function execute() {
         global $DB;
 
-        $cutoff = time() - lib::get_metric_expiry();
+        // We want to use the server's timezone when determining 'midnight'.
+        $tz = \core_date::get_server_timezone_object();
+
+        // Adding 12 hours ensures that we round to the nearest midnight.
+        $secondsago = lib::get_metric_expiry() + (HOURSECS * 12);
```

Comment:
Does it actually round to nearest midnight? what if the expiry was already 1am? then it will round to 12PM

Note, I dont know how expiries are set under the hood

Reply:
The expiry is set via settings. It is in seconds.
According to the PHP docs, 'midnight' sets the time component to '00:00:00', i.e. midnight.
So if the expiry was 2020-02-02 01:00:00, it will get changed to 2020-02-02 13:00:00, and then changed to
2020-02-02 00:00:00.

Reply:
https://www.php.net/manual/en/datetime.formats.relative.php

Reply:
After checking, I discovered that I misinterpreted the docs. I have made adjustments, including extra tests, to ensure that we are indeed rounding to midnight.

Reply:
Also, it has been changed, so that it no longer rounds to the nearest midnight. It always rounds to the previous midnight.